src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -82,6 +82,7 @@ public User actAs(X x, User entity) throws AuthenticationException {
     // Set user and agent objects into the session context and place into sessionDAO.
     Session session = x.get(Session.class);
     session.setUserId(user.getId());
+    session.setUserId(agent.getId());
```

Comment:
Was this supposed to be session.setAgentId() ?

Reply:
Yes, thanks. Fixed by 7be3e47.